appcues/src/main/java/com/appcues/analytics/ExperienceLifecycleTracker.kt

```diff
@@ -65,6 +66,7 @@ internal class ExperienceLifecycleTracker(
                             // if ending on the last step OR an action requested it be considered complete explicitly,
                             // track the experience_completed event
                             trackLifecycleEvent(ExperienceCompleted(it.experience))
+                            onCompleted(it.experience)
```

Comment:
interesting, yeah I'm not quite sure here. I think we need to clear it out of the list on completion or dismiss - any time it goes into EndingExperience state. I think this is what happens on iOS here https://github.com/appcues/appcues-ios-sdk/blob/main/Sources/AppcuesKit/Presentation/ExperienceRendering/ExperienceRenderer.swift#L232

However, I think you noted a point about this - if I leave a screen and the view is torn down, that might trigger a dismiss - but if I return to that screen I might expect the embed to be back, since it wasn't really dismissed by the user? Maybe this is ok, as returning to the screen should trigger a new screen view and qualification in theory? Otherwise, we'd need a way to know about non-modal dismiss that differs from the modal case where we handle the view being destroyed as a dismiss (i.e. a deep link coming in that navigates to another Activity, for example)

Reply:
thanks for bringing that up. not sure what to do here.

Reply:
I think the right thing to do at this point is to remove it from the list on any transition to `EndingExperience` - completed or dismissed.